File: Unidad 3/Meta 3.1/Java/Ecosistema.java
import java.util.*;

public class Ecosistema extends SistemaBiologico {

	private Vector<Biosenosis> biosenosis;
	private Vector<Biotopo> biotopos;

	public Ecosistema(String nombre) {
		super(nombre);
		biosenosis = new Vector<Biosenosis>();
		biotopos = new Vector<Biotopo>();
	}

	public Vector<Biosenosis> getBiosenosis() {
		return biosenosis;
	}

	public void setBiosenosis(Vector<Biosenosis> biosenosis) {
		this.biosenosis = biosenosis;
	}

	public Vector<Biotopo> getBiotopos() {
		return biotopos;
	}

	public void setBiotopos(Vector<Biotopo> biotopos) {
		this.biotopos = biotopos;
	}

	@Override
	public String toString() {
		return "Ecosistema{" + "nombre=" + getNombre() + ", biosenosis=" + biosenosis + ", biotopos=" + biotopos + '}';
	}

}

File: Unidad 3/Meta 3.1/Python/Ecosistema.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
import SistemaBiologico as SistemaBiologico
import Biosenosis as Biosenosis
import Biotopo as Biotopo

from typing import List


class Ecosistema(SistemaBiologico.SistemaBiologico):
	def __init__(self, nombre: str):
		super().__init__(nombre)
		self.biotopos = []
		self.biosenosis = []

	def getBiotopos(self) -> List[Biotopo.Biotopo]:
		return self.biotopos
	
	def setBiotopos(self, biotopos: List[Biotopo.Biotopo]):
		self.biotopos = biotopos

	def getBiosenosis(self) -> List[Biosenosis.Biosenosis]:
		return self.biosenosis

	def setBiosenosis(self, biosenosis: List[Biosenosis.Biosenosis]):
		self.biosenosis = biosenosis

	def __str__(self):
		return "Ecosistema{" + "nombre=" + self.getNombre() + ", biotopos=" + str(self.getBiotopos()) + ", biosenosis=" + str(self.getBiosenosis()) + "}"
